processor/blockprocessor/block_processor.go

```diff
@@ -0,0 +1,78 @@
+package blockprocessor
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/processor"
+)
+
+type blockProcessor struct {
+	handler            func(block *ledgercore.ValidatedBlock) error
+	nextRoundToProcess uint64
+	ledger             *ledger.Ledger
+}
+
+// MakeProcessor creates a block processor
+func MakeProcessor(ledger *ledger.Ledger, handler func(block *ledgercore.ValidatedBlock) error) processor.Processor {
+	return &blockProcessor{ledger: ledger, nextRoundToProcess: 1, handler: handler}
+}
+
+// Process a raw algod block
+func (processor *blockProcessor) Process(cert *rpcs.EncodedBlockCert) error {
+	if processor.ledger == nil {
```

Comment:
I think this check is not needed, but if you really want it, it should probably be in `MakeProcessor()`.